pkg/kfconfig/types_test.go

```diff
@@ -435,6 +439,208 @@ func TestKfConfig_SetSecret(t *testing.T) {
 	}
 }
 
+func TestKfConfig_addPatchStratgicMerge(t *testing.T) {
+	type testCase struct {
+		Input     *types.Kustomization
+		PatchName string
+		Expected  *types.Kustomization
+	}
+
+	cases := []testCase{
+		// New parameter
+		{
+			Input:     &types.Kustomization{},
+			PatchName: "new-config.yaml",
+			Expected: &types.Kustomization{
+				PatchesStrategicMerge: []types.PatchStrategicMerge{
+					types.PatchStrategicMerge("new-config.yaml"),
+				},
+			},
+		},
+		// Existing parameter
+		{
+			Input: &types.Kustomization{
+				PatchesStrategicMerge: []types.PatchStrategicMerge{
+					types.PatchStrategicMerge("config1.yaml"),
+					types.PatchStrategicMerge("exists-config.yaml"),
+					types.PatchStrategicMerge("config2.yaml"),
+				},
+			},
+			PatchName: "exists-config.yaml",
+			Expected: &types.Kustomization{
+				PatchesStrategicMerge: []types.PatchStrategicMerge{
+					types.PatchStrategicMerge("config1.yaml"),
+					types.PatchStrategicMerge("exists-config.yaml"),
+					types.PatchStrategicMerge("config2.yaml"),
+				},
+			},
+		},
+	}
+
+	for _, c := range cases {
+		addPatchStratgicMerge(c.Input, c.PatchName)
+		if !reflect.DeepEqual(c.Input, c.Expected) {
+			pGot, _ := Pformat(c.Input)
+			pWant, _ := Pformat(c.Expected)
+			t.Errorf("Error adding patch %v; got;\n%v\nwant;\n%v", c.PatchName, pGot, pWant)
```

Comment:
I think using `go-cmp` is better for readability.  it's pretty hard to see where went wrong with the literal printout.

Reply:
Done